tests/unit/Entries/GetEntryByFullPathTest.cs

```diff
@@ -0,0 +1,304 @@
+            var expectedUri = new Uri(baseAddress + $"v1/Repositories/repoId/Entries/ByPath?fullPath={urlEncodedRootPath}&fallbackToClosestAncestor=true");
+
+            handlerMock.Protected().Verify(
+               "SendAsync",
+               Times.Exactly(1), // we expected a single external request
+               ItExpr.Is<HttpRequestMessage>(req =>
+                  req.Method == HttpMethod.Get  // we expected a GET request
+                  && req.RequestUri == expectedUri // to this uri
+               ),
+               ItExpr.IsAny<CancellationToken>()
+            );
+        }
+
+        [Fact]
+        public async Task GetEntryByPathAsync_ReturnSuccessfulAncestorStatusCode()
+        {
+            // ARRANGE
+            string baseAddress = "http://api.laserfiche.com/";
+            string repoId = "repoId";
+            string nonExistingPath = @"\NonExistingPath";
+            bool fallBackToClosestAncestor = true;
+            FindEntryResult getAncestorEntryByPath = new FindEntryResult()
+            {
+                Entry = new Entry()
+                {
+                    Id = 1,
+                    Name = "EntryName",
+                    ParentId = 1,
+                    FullPath = @"\",
+                    FolderPath = null,
+                    Creator = "",
+                    CreationTime = DateTimeOffset.Now,
+                    LastModifiedTime = DateTimeOffset.Now,
+                    EntryType = EntryType.Folder,
+                    TemplateName = "Template",
+                    TemplateId = 2,
+                    VolumeName = "Volume"
+                }
+            };
+
+            var handlerMock = new Mock<HttpMessageHandler>(MockBehavior.Strict);
+            handlerMock
+               .Protected()
+               // Setup the PROTECTED method to mock
+               .Setup<Task<HttpResponseMessage>>(
+                  "SendAsync",
+                  ItExpr.IsAny<HttpRequestMessage>(),
+                  ItExpr.IsAny<CancellationToken>()
+               )
+               // prepare the expected response of the mocked http call
+               .ReturnsAsync(new HttpResponseMessage()
+               {
+                   StatusCode = HttpStatusCode.OK,
+                   Content = new StringContent(JsonConvert.SerializeObject(getAncestorEntryByPath))
+               })
+               .Verifiable();
+
+            // use real http client with mocked handler here
+            var httpClient = new HttpClient(handlerMock.Object)
+            {
+                BaseAddress = new Uri(baseAddress),
+            };
+
+            var client = new RepositoryApiClient(httpClient);
+
+            // ACT
+            var result = await client.EntriesClient.GetEntryByPathAsync(repoId, nonExistingPath, fallBackToClosestAncestor);
+
+            // ASSERT   
+            Assert.NotNull(result);
+            Assert.Null(result.AncestorEntry);
```

Comment:
For this test, `GetEntryByPathAsync_ReturnSuccessfulAncestorStatusCode`, shouldn't result.Entry be null instead? Otherwise this is the same as the first test?

Reply:
fixed